mkosi/config.py

```diff
@@ -3763,6 +3775,14 @@ def parse_config_one(path: Path, profiles: bool = False, local: bool = False) ->
             elif hasattr(ParseContext.cli, s.dest):
                 delattr(ParseContext.cli, s.dest)
 
+        env = {}
+
+        for name in getattr(ParseContext.config, "pass_environment", []):
+            if name in getattr(ParseContext.config, "environment", {}):
+                env[name] = ParseContext.config.environment[name]
+
+        setattr(ParseContext.cli, "environment", env)
```

Comment:
Why the `setattr` here instead of just
```suggestion
        ParseContext.cli.environment = env
```